simple-loadflow/src/main/java/com/powsybl/loadflow/simple/network/NetworkContext.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.loadflow.simple.network;
+
+import com.google.common.collect.ImmutableList;
+import com.powsybl.iidm.network.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class NetworkContext {
+
+    private final Network network;
+
+    private final List<Bus> buses;
+
+    private final List<Branch> branches = new ArrayList<>();
+
+    private final Map<String, Bus> busesById;
+
+    private final String slackBusId;
+
+    private final Set<String> pvBusIds = new HashSet<>();
+
+    private final Map<String, Double> busP = new HashMap<>();
+    private final Map<String, Double> busQ = new HashMap<>();
+
+    public NetworkContext(Network network) {
+        this.network = Objects.requireNonNull(network);
+        if (network.getDanglingLineCount() > 0) {
+            throw new UnsupportedOperationException("TODO: dangling lines");
+        }
+        if (network.getThreeWindingsTransformerCount() > 0) {
+            throw new UnsupportedOperationException("TODO: 3 windings transformers");
+        }
+        buses = network.getBusView().getBusStream().filter(Bus::isInMainConnectedComponent).collect(ImmutableList.toImmutableList());
+        busesById = buses.stream().collect(Collectors.toMap(Identifiable::getId, b -> b));
+
+        Map<String, Integer> neighbors = new HashMap<>();
```

Comment:
It looks like it's not used any more, or I missed something!

Reply:
Cleaned